mk/Common.mk

```diff
@@ -232,128 +240,99 @@ endif
+CV_SW_PREFIX = $(TEST_CV_SW_PREFIX)
+else
+ifdef  CFG_CV_SW_PREFIX
+CV_SW_PREFIX = $(CFG_CV_SW_PREFIX)
+else
+$(error CV_SW_PREFIX not defined in either the shell environment, test.yaml or cfg.yaml)
+endif
+endif
 endif
 
-ifeq ($(COREV_YES),YES)
-ifeq ($(call IS_YES,$(TEST_COREV_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the COREV toolchain)
+ifndef CV_SW_MARCH
+ifdef  TEST_CV_SW_MARCH
+CV_SW_MARCH = $(TEST_CV_SW_MARCH)
+else
+ifdef  CFG_CV_SW_MARCH
+CV_SW_MARCH = $(CFG_CV_SW_MARCH)
+else
+CV_SW_MARCH = rv32imc
+$(warning CV_SW_MARCH not defined in either the shell environment, test.yaml or cfg.yaml)
+endif
 endif
-RISCV            = $(COREV_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(COREV_VENDOR)-elf-
-RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(COREV_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_COREV_MARCH),$(COREV_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_COREV_CFLAGS),$(COREV_CFLAGS))
 endif
 
-ifeq ($(PULP_YES),YES)
-ifeq ($(call IS_YES,$(TEST_PULP_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the PULP toolchain)
+ifndef CV_SW_CC
+ifdef  TEST_CV_SW_CC
+CV_SW_CC = $(TEST_CV_SW_CC)
+else
+ifdef  CFG_CV_SW_CC
+CV_SW_CC = $(CFG_CV_SW_CC)
+else
+CV_SW_CC = gcc
+$(warning CV_SW_CC not defined in either the shell environment, test.yaml or cfg.yaml)
+endif
 endif
-RISCV            = $(PULP_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(PULP_VENDOR)-elf-
-RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(PULP_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_PULP_MARCH),$(PULP_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_PULP_CFLAGS),$(PULP_CFLAGS))
 endif
 
-ifeq ($(LLVM_YES),YES)
-ifeq ($(call IS_YES,$(TEST_LLVM_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the LLVM toolchain)
+ifndef CV_SW_FLAGS
+ifdef  TEST_CV_SW_FLAGS
+CV_SW_FLAGS = $(TEST_CV_SW_FLAGS)
+else
+ifdef  CFG_CV_SW_FLAGS
+CV_SW_FLAGS = $(CFG_CV_SW_FLAGS)
+else
+$(warning CV_SW_FLAGS not defined in either the shell environment, test.yaml or cfg.yaml)
+endif
 endif
-RISCV            = $(LLVM_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(LLVM_VENDOR)-elf-
-RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(LLVM_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_LLVM_MARCH),$(LLVM_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_LLVM_CFLAGS),$(LLVM_CFLAGS))
-RISCV_CFLAGS    += -menable-experimental-extensions
 endif
 
-CFLAGS ?= -Os -g -static -mabi=ilp32 -march=$(RISCV_MARCH) -Wall -pedantic
+RISCV            = $(CV_SW_TOOLCHAIN)
+RISCV_PREFIX     = $(CV_SW_PREFIX)
+RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
 
-ifeq ($(firstword $(subst _, ,$(TEST))),pulp)
-  CFLAGS = -Os -g -D__riscv__=1 -D__LITTLE_ENDIAN__=1 -march=rv32imcxpulpv2 -Wa,-march=rv32imcxpulpv2 -fdata-sections -ffunction-sections -fdiagnostics-color=always
-endif
+RISCV_MARCH      = $(CV_SW_MARCH)
+RISCV_CC         = $(CV_SW_CC)
+RISCV_FLAGS      = $(CV_SW_FLAGS)
```

Comment:
Should this be `RISCV_CFLAGS = $(CV_SW_CFLAGS)`?
And should there be TEST and CFG variants of it too?

Reply:
You are right.  I have changed all instances of `FLAGS` to `CFLAGS` and also added `CV_SW_CFLAGS` to the table of "Optional COREV Environment Variables"

Reply:
Whoops, I _thought_ I had done this.  Pushed it in just now.